plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -3110,7 +3110,7 @@ private long getStaticMax(final String os, final boolean b, final long dynamicMi
             s_logger.warn("No recommended value found for dynamic max, setting static max and dynamic max equal");
             return dynamicMaxRam;
         }
-        final long staticMax = Math.min(recommendedValue, 4l * dynamicMinRam); // XS
+        final long staticMax = Math.max(recommendedValue, 4L * dynamicMinRam); // XS
```

Comment:
4 times dynamicMinRam can be higher than the recommended value, so it needs be min() only.